packages/client/src/LogionClient.ts

```diff
@@ -0,0 +1,61 @@
+import { AuthenticatedLogionClient } from "./AuthenticatedLogionClient";
+import { AccountTokens } from "./AuthenticationClient";
+import { ComponentFactory, DefaultComponentFactory } from "./ComponentFactory";
+import { LogionClientConfig, SharedState } from "./SharedClient";
+import { RawSigner } from "./Signer";
+import { LegalOfficer } from "./Types";
+
+export class LogionClient {
+
+    static create(config: LogionClientConfig): LogionClient {
+        const componentFactory = getComponentFactory(config);
+        const axiosFactory = componentFactory.buildAxiosFactory();
+        const sharedState = {
+            config,
+            componentFactory,
+            axiosFactory,
+            directoryClient: componentFactory.buildDirectoryClient(config.directoryEndpoint, axiosFactory),
+        };
+        return new LogionClient(sharedState);
+    }
+
+    constructor(sharedState: SharedState) {
+        this.sharedState = sharedState;
+    }
+
+    private sharedState: SharedState;
+
+    async getLegalOfficers(): Promise<LegalOfficer[]> {
+        return this.sharedState.directoryClient.getLegalOfficers();
+    }
+
+    async authenticate(addresses: string[], signer: RawSigner): Promise<AuthenticatedLogionClient> {
+        const legalOfficers = await this.getLegalOfficers();
+        const client = this.sharedState.componentFactory.buildAuthenticationClient(this.sharedState.config.directoryEndpoint, legalOfficers, this.sharedState.axiosFactory);
+        const tokens = await client.authenticate(addresses, signer);
+        return new AuthenticatedLogionClient({
+            sharedState: this.sharedState,
+            legalOfficers,
+            tokens,
+            currentAddress: addresses[0]
+        });
+    }
+
+    async useTokens(tokens: AccountTokens): Promise<AuthenticatedLogionClient> {
+        const legalOfficers = await this.sharedState.directoryClient.getLegalOfficers();
+        return new AuthenticatedLogionClient({
+            sharedState: this.sharedState,
+            legalOfficers,
+            tokens,
+            currentAddress: tokens.addresses[0],
```

Comment:
This method should select the token matching address given as param, rather than always returning the first one ?

Reply:
The address is not given as param, that's why an arbitrary address is selected. The flow would be as follows:

```
const client = LogionClient.create(...);
const authenticated = await client.useTokens(...).withCurrentAddress(...);
```